Examples/DicomConvert/DicomConvert.py

```diff
@@ -0,0 +1,110 @@
+#=========================================================================
+#
+#  Copyright NumFOCUS
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0.txt
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#=========================================================================
+
+import SimpleITK as sitk
+import multiprocessing
+import argparse
+import os
+import sys
+import functools
+import pandas as pd
+import itertools
+
+def convert_image(input_file_name, output_file_name, new_width=None):
+    try:
+        image_file_reader = sitk.ImageFileReader()
+        # only read DICOM images
+        image_file_reader.SetImageIO('GDCMImageIO')
+        image_file_reader.SetFileName(input_file_name)
+        image = image_file_reader.Execute()[:,:,0]
+        if new_width:
+            original_size = image.GetSize()
+            original_spacing = image.GetSpacing()
+            new_spacing = [(original_size[0]-1)*original_spacing[0]/(new_width-1)]*2
+            new_size = [new_width, int((original_size[1]-1)*original_spacing[1]/new_spacing[1])]
+            image = sitk.Resample(image, new_size, sitk.Transform(), sitk.sitkLinear,
+                                  image.GetOrigin(), new_spacing, image.GetDirection(),0,
+                                  image.GetPixelID())
+        # If a single channel image, rescale to [0,255]. Also modify the intensity values based
+        # on the photomertic interpretation. If MONOCHROME2 (minimum should be displayed as black) we
+        # don't need to do anything, if image has MONOCRHOME1 (minimum should be displayed as white) we flip
+        # the intensities. This is a constraint imposed by ITK which always assumes MONOCHROME2.
+        if image.GetNumberOfComponentsPerPixel() == 1:
+            image = sitk.RescaleIntensity(image, 0, 255)
+            if image_file_reader.GetMetaData('0028|0004').strip() == 'MONOCHROME1':
```

Comment:
With the above IO region change, the meta-data may be in the image, if the resample hasn't been done. Perhaps the rescale intensities should come first? I'm not sure if the inverting the default pixel value would have and effect.

Reply:
The meta-data is lost in 95% of the image types (jpg,png...), so not really a concern.